usaspending_api/references/v2/views/cfda.py

```diff
@@ -57,7 +64,13 @@ def _populate_cfdas_if_needed(self):
             CFDA_DICTIONARY = response
 
     def _request_from_grants_api(self):
-        return post(
-            "https://www.grants.gov/grantsws/rest/opportunities/search/cfda/totals",
+        cfda_response = post(
+            CFDA_URL,
             headers={"Authorization": f"APIKEY={settings.GRANTS_API_KEY}"},
-        ).json()["cfdas"]
+        )
+        if cfda_response.status_code == status.HTTP_503_SERVICE_UNAVAILABLE:
```

Comment:
Perhaps check if the status code is not a 200? If not 200 then log the status code and response and return either an empty array or raise NoDataFoundException